File: leetcode/2478-longest-nice-subarray/solution.c
int longestNiceSubarray(int* nums, int n) {
    int ret = 0;
    int j = 0;
    int sum = 0;
    for (int i = 0; i < n; i++) {
        while((nums[i] & sum) != 0) sum &= ~nums[j++];
        int score = i-j+1;
        if(score > ret) ret = score;
        sum |= nums[i];
    }
    return ret;
}

File: leetcode/2478-longest-nice-subarray/solution.rs
impl Solution {
    pub fn longest_nice_subarray(nums: Vec<i32>) -> i32 {
        let mut ret = 0;
        let mut j = 0;
        let mut sum = 0;
        for i in 0..nums.len() {
            while nums[i] & sum != 0 {
                sum &= !nums[j];
                j += 1;
            }
            ret = ret.max(i-j+1);
            sum |= nums[i];
        }
        return ret as i32;
    }
}
